apps/server/src/shared/domain/rules/team.rule.spec.ts

```diff
@@ -33,35 +30,65 @@ describe('TeamRule', () => {
+		const teamRole1 = roleFactory.buildWithId({ permissions: [teamPermissionA] });
+		const teamRole2 = roleFactory.buildWithId({ permissions: [teamPermissionB], roles: [teamRole1] });
+		const teamRole = roleFactory.buildWithId({ permissions: [teamPermissionC], roles: [teamRole2] });
+		const user = userFactory.buildWithId({ roles: [role] });
+		const entity = teamFactory.withRoleAndUserId(teamRole, user.id).build();
+		return {
+			role,
+			teamRole1,
+			teamRole2,
+			teamRole,
+			user,
+			entity,
+		};
+	};
 
 	describe('isApplicable', () => {
 		it('should return truthy', () => {
-			expect(() => service.isApplicable(entity.teamUsers[0].user, entity)).toBeTruthy();
+			const { user, entity } = setup();
+			expect(() => service.isApplicable(user, entity)).toBeTruthy();
 		});
 	});
 
 	describe('hasPermission', () => {
-		it('should return "true" if user in scope', () => {
-			const res = service.hasPermission(entity.teamUsers[0].user, entity, PermissionContextBuilder.read([permissionB]));
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionA]));
 			expect(res).toBe(true);
 		});
 
-		it('should return "false" if user has not permission', () => {
-			const res = service.hasPermission(entity.teamUsers[0].user, entity, PermissionContextBuilder.read([permissionC]));
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionB]));
+			expect(res).toBe(true);
+		});
+
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionC]));
+			expect(res).toBe(true);
+		});
+
+		it('should return "false" if user in scope but without permission', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionD]));
+			expect(res).toBe(false);
+		});
+
+		it('should return "false" if user has global permission', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([permissionA]));
 			expect(res).toBe(false);
 		});
 
-		it('should throw error if entity was not found', () => {
-			const notFoundUser = userFactory.build({ roles: [role] });
-			expect(() => service.hasPermission(notFoundUser, entity, PermissionContextBuilder.read([permissionA]))).toThrow(
-				new InternalServerErrorException('Cannot find user in team')
-			);
+		it('should return "false" if user has not permission', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([permissionC]));
+			expect(res).toBe(false);
 		});
 	});
 });
```

Comment:
You should add a test case for a user that is not in the team.

Reply:
created